src/Microsoft.AspNet.Mvc.Core/ModelBinding/CompositeModelBinder.cs

```diff
@@ -203,5 +209,21 @@ private static BodyBindingState GetBodyBindingState(ModelBindingContext oldBindi
 
             return state;
         }
+
+        private static ModelValidationNode GetModelValidationNode(
+            string key,
+            ModelBindingResult result,
+            ModelMetadata metadata)
+        {
+            if (result.IsModelSet && result.ValidationNode == null)
+            {
+                return new ModelValidationNode(
+                    key,
+                    metadata,
+                    result.Model);
+            }
+
+            return result.ValidationNode;
+        }
```

Comment:
IMO it would be much less confusing to have this code **inline** and have a comment explaining under what conditions this might occur. These are kinds of interactions we want to highlight, not hide.